src/app/wallets/pay-on-chain.ts

```diff
@@ -153,59 +153,56 @@ const executePaymentViaIntraledger = async ({
   logger,
 }: {
   senderAccount: Account
-  senderWalletId: WalletId
+  senderWallet: Wallet
   recipientWallet: Wallet
   amount: Satoshis
   address: OnChainAddress
   memo: string | null
   sendAll: boolean
   logger: Logger
 }): Promise<PaymentSendStatus | ApplicationError> => {
-  if (recipientWallet.id === senderWalletId) return new SelfPaymentError()
+  if (recipientWallet.id === senderWallet.id) return new SelfPaymentError()
 
   const intraledgerLimitCheck = await checkIntraledgerLimits({
     amount,
-    walletId: senderWalletId,
+    walletId: senderWallet.id,
     account: senderAccount,
   })
   if (intraledgerLimitCheck instanceof Error) return intraledgerLimitCheck
 
   const usdPerSat = await getCurrentPrice()
   if (usdPerSat instanceof Error) return usdPerSat
 
-  const fee = toSats(0)
-  const sats = toSats(amount + fee)
-  const usd = sats * usdPerSat
-  const usdFee = fee * usdPerSat
+  const amountDisplayCurrency = DisplayCurrencyConversionRate(usdPerSat)(amount)
 
   const recipientAccount = await AccountsRepository().findById(recipientWallet.accountId)
   if (recipientAccount instanceof Error) return recipientAccount
 
   return LockService().lockWalletId(
-    { walletId: senderWalletId, logger },
+    { walletId: senderWallet.id, logger },
     async (lock) => {
-      const balance = await LedgerService().getWalletBalance(senderWalletId)
+      const balance = await LedgerService().getWalletBalance(senderWallet.id)
       if (balance instanceof Error) return balance
-      if (balance < sats)
+      if (balance < amount)
         return new InsufficientBalanceError(
-          `Payment amount '${sats}' exceeds balance '${balance}'`,
+          `Payment amount '${amount}' exceeds balance '${balance}'`,
         )
 
       const onchainLoggerOnUs = logger.child({ balance, onUs: true })
       const journal = await LockService().extendLock(
         { logger: onchainLoggerOnUs, lock },
         async () =>
-          LedgerService().addOnChainIntraledgerTxTransfer({
-            senderWalletId,
+          LedgerService().addOnChainIntraledgerTxSend({
```

Comment:
Looks like we lost the `addOnChainIntraledgerTxTransfer` change somehow in this diff